src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -87,6 +87,76 @@
+     */
+    public EmailWrapper generateFeedbackSessionResendSummaryOfCourse(String courseId, StudentAttributes student) {
+        
+        CourseAttributes course = coursesLogic.getCourse(courseId);
+        
+        List<FeedbackSessionAttributes> sessions = new ArrayList<FeedbackSessionAttributes>();
+        List<FeedbackSessionAttributes> fsInCourse = fsLogic.getFeedbackSessionsForCourse(courseId);
+        
+        for (FeedbackSessionAttributes fsa : fsInCourse) {
+            if (!fsa.isPrivateSession() && (fsa.isSentOpenEmail() || fsa.isSentPublishedEmail())) {
+                sessions.add(fsa);
+            }
+        }
+        
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String joinUrl = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+        
+        String joinFragmentValue = isYetToJoinCourse(student)
+                                   ? Templates.populateTemplate(EmailTemplates.FRAGMENT_STUDENT_COURSE_JOIN,
+                                           "${joinUrl}", joinUrl,
+                                           "${courseName}", course.getName())
+                                   : "";
+        
+        for (FeedbackSessionAttributes fsa : sessions) {
+            String submitUrl = fsa.isOpened()
+                        ? "<a href = \"" + Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(fsa.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString() + "\">link<//a>"
+                        : "(Feedback session is " + (fsa.isClosed() ? "closed" : "not yet opened") + ")";
+
+            String reportUrl = fsa.isPublished()
+                        ? "<a href = \"" + Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
+                                .withCourseId(course.getId())
+                                .withSessionName(fsa.getFeedbackSessionName())
+                                .withRegistrationKey(StringHelper.encrypt(student.key))
+                                .withStudentEmail(student.email)
+                                .toAbsoluteString() + "\">link<//a>"
+                        : "(Feedback session is not yet published)";
+            
+            linksFragmentValue.append(Templates.populateTemplate(
+                    EmailTemplates.FRAGMENT_SINGLE_FEEDBACK_SESSION_LINKS,
+                    "${feedbackSessionName}", fsa.getFeedbackSessionName(),
+                    "${deadline}", fsa.isClosed() ? TimeHelper.formatTime12H(fsa.getEndTime()) + " (Passed)"
+                                                  : TimeHelper.formatTime12H(fsa.getEndTime()),
```

Comment:
To avoid repetition, `TimeHelper.formatTime12H(fsa.getEndTime()) + (fsa.isClosed() ? " (Passed)" : "")`